url9ba68db77a_samkio_Dungeonator_tgz-pJ8-net_woopa_dungeon_core_Main2J8/src/net/woopa/dungeon/datatypes/Grid.java

```diff
@@ -0,0 +1,182 @@
+package net.woopa.dungeon.datatypes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.woopa.dungeon.core.CoreMaterial;
+import net.woopa.dungeon.managers.MaterialManager;
+
+public class Grid {
+	private final Vector2D size;
+	private Material[][] area;
+	private int used = 0;
+
+	public Grid(int sx, int sy) {
+		this.size = new Vector2D(sx, sy);
+		area = new Material[size.getX()][size.getY()];
+		for (int x = 0; x < size.getX(); x++) {
+			for (int y = 0; y < size.getY(); y++) {
+				area[x][y] = CoreMaterial.UNDEF;
+			}
+		}
+	}
+
+	public Grid(String[] strings) {
+		this(strings[0].length(), strings.length);
+		for (int y = 0; y < size.getY(); y++) {
+			for (int x = 0; x < size.getX(); x++) {
+				char c = strings[size.getY() - y - 1].charAt(x);
+				area[x][y] = MaterialManager.getMaterial(c);
+			}
+		}
+	}
+
+	public Grid(Vector2D levelSize) {
+		this(levelSize.getX(), levelSize.getY());
+	}
+
+	public float percentUtilized() {
+		return ((float) used * 100) / ((float) size.getX() * size.getY());
+	}
+
+	public void show() {
+		for (String s : getMap()) {
+			System.out.println(s);
+		}
+	}
+
+	public List<String> getMap() {
+		List<String> list = new ArrayList<String>();
+		String line = "";
+		for (int x = 0; x < size.getX() + 2; x++) {
+			line += CoreMaterial.NONE.getSymbol();
+		}
+		list.add(line);
+		for (int y = size.getY() - 1; y >= 0; y--) {
+			line = "";
+			line += CoreMaterial.NONE.getSymbol();
+			for (int x = 0; x < size.getX(); x++) {
+				Material b = area[x][y];
+				line += b.getSymbol();
+			}
+			line += CoreMaterial.NONE.getSymbol();
+			list.add(line);
+
+		}
+		line = "";
+		for (int x = 0; x < size.getX() + 2; x++) {
+			line += CoreMaterial.NONE.getSymbol();
+		}
+		list.add(line);
+		return list;
+	}
+
+	public Vector2D getSize() {
+		return this.size;
+	}
+
+	public void renderBasicEmptyRoom(int origin_x, int origin_y, int size_x,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.